include/dali/core/cuda_stream_pool.h

```diff
@@ -141,13 +141,38 @@ class CUDAStreamLease {
     reset();
   }
 
+  /**
+   * @brief Obtains the leased stream handle
+   */
+  cudaStream_t get() const & noexcept {
+    return stream_;
+  }
+
+  /**
+   * @brief Cannot obtain a valie handle from a temporary CUDAStreamLease
+   *
+   * By the time this function returns, the stream is returned to the pool.
```

Comment:
I wonder, if we are at this, we should probably mention, that the user of this API should keep the StreamLease for the duration the stream is used.

Reply:
I think that's what the word "lease" means, doesn't it?

Reply:
The class description describes the object life cycle.